src/transformers/dynamic_module_utils.py

```diff
@@ -143,9 +144,24 @@ def get_class_in_module(class_name, module_path):
     """
     Import a module on the cache directory for modules and extract a class from it.
     """
-    module_path = module_path.replace(os.path.sep, ".")
-    module = importlib.import_module(module_path)
-    return getattr(module, class_name)
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        module_dir = Path(HF_MODULES_CACHE) / os.path.dirname(module_path)
+        module_file_name = module_path.split(os.path.sep)[-1] + ".py"
+
+        # copy to a temporary directory
+        shutil.copy(f"{module_dir}/{module_file_name}", tmp_dir)
+        cmd = f'import os; os.remove("{module_dir}/{module_file_name}")'
+        os.system(f"python3 -c '{cmd}'")
```

Comment:
no more test error is we remove the file in a subprocess.

Reply:
That's a bit crazy! Can you use the subprocess command instead of `os.system`? Not sure if this is going to fly well on Windows for instance.

Reply:
Changed to `subprocess`. Tested on my Windows env. and it works.